### ISSUE TITLE
Serving OEO classes: Readable by humans and machines

### ISSUE THREAD

**Issue by l-emele** [OpenEnergyPlatform/oeplatform#763]
Via the OEO DEV mailing list we got the following feedback for the OEO:
> Unfortunately the OEO is still not usable (for Linked Data). Just have a look at http://openenergy-platform.org/ontology/oeo/OEO_00230002 - nothing there. I hope you now understand that it is crucial, to return sth. machine and human readable for every OEO id in order to be recognized and accepted as an ontology and you now also understand the direct practical reasons for it when it comes to actually using the ontology when using this "human-incomprehensible" IDs.

So it basically this asks that the class IRIs resolve to some kind of webpages for the specific pages and not just serving the content of http://openenergy-platform.org/ontology/oeo/

We discussed that today in the OEO DEV meeting. This is not a problem of the ontology itself, but if we want to have the functionality of serving the classes it would be a topic for the OEP. This should be discussed at the OEP DEV and/or the LOD_GEOSS meetings.

**Comment by l-emele** [comment]
@christian-rli : I assigned this issue to you are both the bridge between the OEO and OEP and the bridge between SIROP and LOD_GEOSS.

**Comment by christian-rli** [comment]
@Adel-Mem this is the issue we talked about in the OEP-DEV meeting. See how far you can get and we'll talk about it in the next meeting. Thank you for volunteering :)

**Comment by adelmemariani** [comment]
I added the functionality to display sub/super classes for a given class. I think we should add the names for the classes. Also if it is needed, I can work on a new user interface to show the classes in a tree or graph view.

**Comment by l-emele** [comment]
A tree or graph view would be imho great!

**Comment by han-f** [comment]
I agree with @l-emele, a tree or graph view would be a great value added!

**Comment by han-f** [comment]
Is this the correct issue to post the ideas we had in yesterdays (28/07/21) SIROP Jour Fixe when @adelmemariani showed us the progress that he has made on displaying OEOs terms on the OEP  (great value added!)? 
Here are the ideas from yesterday: 

- For each term its definition should be displayed
- Further items that should be displayed best to be discussed in an upcoming oeo-dev meeting.

**Comment by adelmemariani** [comment]
Based on the new [oeo_viewer](https://github.com/OpenEnergyPlatform/oeo-viewer) project and the above commits, I think we can have two views: 1- Text-based folding menus. 2- Graph and tree views.  As it is shown in the following image, if we use the class ID in the URL, the viewers specifically, show the requested class. In case some classes do not have enough description or editor notes, the user can use the 'Parent' button to see the parent and read the parent's description. The 'shrink' and 'expand' buttons help the user to navigate the ontology classes. 
<img width="1792" alt="Viewer" src="https://user-images.githubusercontent.com/7637364/131710334-48271f9b-326c-4bd2-a514-37110fe83d60.png">

**Comment by stap-m** [comment]
It would be great if `alternative terms` would also be shown under the defitition and would be searchable in the search bar. Is that doable @adelmemariani ?

**Comment by adelmemariani** [comment]
> It would be great if `alternative terms` would also be shown under the defitition and would be searchable in the search bar. Is that doable @adelmemariani ?

Yes, it is doable. Basically, everything can be searchable :) 
Also, `example usage` should be there.
I will work on it after I finish my current tasks. So it would not be available for the next release in October...

**Comment by adelmemariani** [comment]
A new clickable layout for the sub-class relations in the OEO:

![Screenshot 2022-04-01 at 17 16 52](https://user-images.githubusercontent.com/7637364/161294132-4ebab219-ea1b-4bae-9c8d-49ef588535ec.png)
![Screenshot 2022-04-01 at 17 17 11](https://user-images.githubusercontent.com/7637364/161294151-e4b4b455-7aaf-4ea6-b32b-85a2b2fab1ef.png)
![Screenshot 2022-04-01 at 17 17 55](https://user-images.githubusercontent.com/7637364/161294179-4b05e465-fd60-4bb3-892e-8a36
![Screenshot 2022-04-01 at 17 20 39](https://user-images.githubusercontent.com/7637364/161294198-18d9bb8f-3456-4e88-9a77-02edd6b22819.png)
d3fe957a.png)

**Comment by l-emele** [comment]
I cannot imagine how such a radar will look, once also the inferred classes will be shown and the structure is not mono-hierarchically any more.

**Comment by adelmemariani** [comment]
@l-emele you are right. That was also my concern. The layout only works for hierarchical relations i.e., sub-classes. I am still thinking about a new layout for relations that are not hierarchical.

**Comment by jh-RLI** [comment]
With the latest version, I have made some fundamental revisions to the oeo pages. The descriptions are now visible and the structure of the page is more consistent with the OEP & clearer. In addition, the navigation between the pages or between the OEO classes is now a little easier. There is still more room for improvement here. 
I would be interested to know what other information should be displayed on the pages.

There is another PR that will improve the user experience by improving the initial page response and indicating that something is loading. 

This issue is now a collection of many ideas, we should endeavour to create issues that tackle parts of it, otherwise things get lost in the void. 

Also see: https://github.com/OpenEnergyPlatform/oeplatform/issues/1496
